### DISCUSSION TITLE
How to use transaction (examples)?

### DISCUSSION THREAD

**Discussion by ETretyakov** [encode/databases#402] [Q&A] [answered]
Could someone provide examples on how to use transactions? There are incomplete examples in documentation but could you provide more exact examples?

```python
async with database.transaction():
    [what is here?]
```

```python
transaction = await database.transaction()
try:
    await transaction.start()
    [what is here?]
except:
    await transaction.rollback()
else:
    await transaction.commit()
```

**Comment by aminalaee** [comment, marked as answer]
Hi,
First off, the example you see in the docs is invalid. That will be fixed when [0.5.3](https://github.com/encode/databases/pull/398) is merged. You don't need to call `await tranasaction.start()`.

```python
transaction = await database.transaction()
try:
    [INSERT/UPDATE] inside that transaction....
except:
    await transaction.rollback()
else:
    await transaction.commit()
```

You can do INSERT or UPDATE within that transaction, and commit/rollback based on that.

**Reply by mathans1695** [reply]
How do we perform insert/update using transaction? Can we do like await database.execute(query=query) insert try except block. I tried that but rollback is not happening if error occurred
